src/mono/mono.proj

```diff
@@ -933,16 +938,52 @@
     <Error Condition="!Exists('$(LocalDotnet)')" Text="'$(LocalDotnet)' doesn't exist." />
   </Target>
 
+  <Target Name="BuildMonoRuntimeCMake" Condition="'$(MonoEnableCMake)' == 'true'">
+    <Exec Command="which ninja" IgnoreExitCode="true" IgnoreStandardErrorWarningFormat="true">
+      <Output TaskParameter="ExitCode" PropertyName="MonoNinjaFound"/>
+    </Exec>
+    <PropertyGroup>
+      <MonoBuildTool Condition="'$(MonoNinjaFound)' == '0'">ninja</MonoBuildTool>
+      <MonoBuildTool Condition="'$(MonoNinjaFound)' != '0'">make -j$([System.Environment]::ProcessorCount)</MonoBuildTool>
+    </PropertyGroup>
+    <ItemGroup>
+      <MonoCMakeArgs Condition="'$(MonoBuildTool)' == 'ninja'" Include="-G Ninja"/>
+      <MonoCMakeArgs Include="-DCMAKE_INSTALL_PREFIX=$(MonoObjDir)out"/>
+    </ItemGroup>
+    <ItemGroup Condition="'$(TargetsOSX)' == 'true'">
+      <MonoCMakeArgs Include="-DCMAKE_C_FLAGS=-mmacosx-version-min=10.12"/>
```

Comment:
we already have a property for the min OSX version we support:
```suggestion
      <MonoCMakeArgs Include="-DCMAKE_C_FLAGS=-mmacosx-version-min=$(macOSVersionMin)"/>
```

that said, it looks like we're missing this for the autotools-based version too. I'll send a PR to fix that.

Reply:
alternatively I think the preferred way to set this in CMake is to use `-DCMAKE_OSX_DEPLOYMENT_TARGET=10.13`